bee-network/src/service/service.rs

```diff
@@ -0,0 +1,485 @@
+
+            info!("Network service started.");
+
+            Ok(Self::default())
+        }
+    }
+}
+
+pub mod standalone {
+    use super::*;
+
+    pub struct NetworkService {
+        pub shutdown: oneshot::Receiver<()>,
+    }
+
+    impl NetworkService {
+        pub fn new(shutdown: oneshot::Receiver<()>) -> Self {
+            Self { shutdown }
+        }
+
+        pub async fn start(self, config: NetworkServiceConfig) {
+            let NetworkService { shutdown } = self;
+            let NetworkServiceConfig {
+                local_keys: _,
+                senders,
+                receivers,
+                peerlist,
+            } = config;
+
+            let Receivers {
+                commands,
+                internal_events,
+            } = receivers;
+
+            let (shutdown_tx1, shutdown_rx1) = oneshot::channel::<()>();
+            let (shutdown_tx2, shutdown_rx2) = oneshot::channel::<()>();
+            let (shutdown_tx3, shutdown_rx3) = oneshot::channel::<()>();
+
+            tokio::spawn(async move {
+                shutdown.await.expect("receiving shutdown signal");
+
+                shutdown_tx1.send(()).expect("receiving shutdown signal");
+                shutdown_tx2.send(()).expect("receiving shutdown signal");
+                shutdown_tx3.send(()).expect("receiving shutdown signal");
+            });
+            tokio::spawn(command_processor(
+                shutdown_rx1,
+                commands,
+                senders.clone(),
+                peerlist.clone(),
+            ));
+            tokio::spawn(event_processor(
+                shutdown_rx2,
+                internal_events,
+                senders.clone(),
+                peerlist.clone(),
+            ));
+            tokio::spawn(peer_checker(shutdown_rx3, senders, peerlist));
+
+            info!("Network service started.");
+        }
+    }
+}
+
+async fn command_processor(shutdown: Shutdown, commands: CommandReceiver, senders: Senders, peerlist: PeerList) {
+    debug!("Command handler running.");
+
+    let mut commands = ShutdownStream::new(shutdown, UnboundedReceiverStream::new(commands));
+
+    while let Some(command) = commands.next().await {
+        if let Err(e) = process_command(command, &senders, &peerlist).await {
+            error!("Error processing command. Cause: {}", e);
+            continue;
+        }
+    }
+
+    debug!("Command processor stopped.");
+}
+
+async fn event_processor(shutdown: Shutdown, events: InternalEventReceiver, senders: Senders, peerlist: PeerList) {
+    debug!("Event handler running.");
+
+    let mut int_events = ShutdownStream::new(shutdown, UnboundedReceiverStream::new(events));
+
+    while let Some(int_event) = int_events.next().await {
+        if let Err(e) = process_internal_event(int_event, &senders, &peerlist).await {
+            error!("Error processing internal event. Cause: {}", e);
+            continue;
```

Comment:
I'm confused by the purpose of this continue.

Reply:
It isn't necessary atm because the Ok case essentially also just continues atm. But maybe me (or someone else) wants to change that in the future. Then the Err case would print the `error` log and enter the execution branch of the Ok case. I like it how it is where it reads almost like a sentence "If there is an error (ignore the rest of the loop body), and straigt continue with the next event"